ipc/glue/ProtocolUtils.cpp

```diff
@@ -517,6 +517,13 @@ void IProtocol::SetManagerAndRegister(IProtocol* aManager, int32_t aId) {
 
 bool IProtocol::ChannelSend(IPC::Message* aMsg) {
   UniquePtr<IPC::Message> msg(aMsg);
+
+  // We don't have a way to send IPC messages at non-deterministic points.
+  // For now, silently discard the messages.
+  if (recordreplay::AreThreadEventsDisallowed()) {
+    return true;
```

Comment:
Why is this OK?

Reply:
This will cause resources in the UI process to leak, but we don't have any alternative right now.  The size of the leaks should be pretty small, I don't see excessive memory usage in either the UI or content processes in a 10 minute recording.